lib/beaker/shared/subcommands_util.rb

```diff
@@ -30,7 +38,72 @@ def reset_argv(args)
       # @param [Array<String>] args the arguments determined by a specific subcommand
       def execute_beaker(*args)
         reset_argv(args)
-        Beaker::CLI.new.execute! 
+        Beaker::CLI.new.execute!
+      end
+
+      # Checks for the presence of a rake file, and creates a rake file if no rake file is present
+      # @return [String] the name of the rakefile to use
+      def determine_rake_file()
+        RAKE_FILES.each do |rake_file|
+          return rake_file if File.file?(rake_file)
+        end
+        File.open RAKE_FILES.first, "w"
```

Comment:
So if the file exists, we won't reach the line where I write to the file, because I return the file name on line 48 if it exists. So if we've got to line 50, it's because no rakefile currently exists, and I'm writing to it just to create an empty file, but that's actually redundant because I append to it later in the code run with "a+" which creates the file if it doesn't exist, so really, I should just be returning the name of the Rakefile here, not writing to it. 👍